API.md

```diff
@@ -0,0 +1,552 @@
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Connection>` | Promise resolves with the [Connection](https://github.com/libp2p/js-interfaces/tree/master/src/connection) instance |
+
+#### Example
+
+```js
+// ...
+const conn = await libp2p.dial(remotePeerInfo)
+```
+
+### dialProtocol
+
+Dials to another peer in the network and selects a protocol to communicate with that peer.
+
+`dialProtocol(peer, protocols, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peer | [PeerInfo](https://github.com/libp2p/js-peer-info), [PeerId](https://github.com/libp2p/js-peer-id), [multiaddr](https://github.com/multiformats/js-multiaddr), `string` | peer to dial |
+| protocols | `String|Array<String>` |  A list of protocols (or single protocol) to negotiate with. Protocols are attempted in order until a match is made. (e.g '/ipfs/bitswap/1.1.0') |
+| [options] | `Object` | dial options |
+| [options.signal] | [`AbortSignal`](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) | An `AbortSignal` instance obtained from an [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) that can be used to abort the connection before it completes |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Connection>` | Promise resolves with the [Connection](https://github.com/libp2p/js-interfaces/tree/master/src/connection) instance |
+
+#### Example
+
+```js
+// ...
+const conn = await libp2p.dialProtocol(remotePeerInfo, protocols)
+```
+
+### hangUp
+
+Closes an open connection with a peer, graciously.
+
+`hangUp(peer)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| peer | [PeerInfo](https://github.com/libp2p/js-peer-info), [PeerId](https://github.com/libp2p/js-peer-id), [multiaddr](https://github.com/multiformats/js-multiaddr), `string` | peer to hang up |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<void>` | Promise resolves once connection closes |
+
+#### Example
+
+```js
+// ...
+await libp2p.hangUp(remotePeerInfo)
+```
+
+### handle
+
+Registers a given handler for the given protocols.
+
+`libp2p.handle(protocols, handler)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| protocols | `Array<String>|String` | protocols to register |
+| handler | `function({ connection:*, stream:*, protocol:string })` | handler to call |
+
+
+#### Example
+
+```js
+// ...
+const handler = ({ connection, stream, protocol }) => {
+  // use stream or connection according to the needs
+}
+
+libp2p.handle('/echo/1.0.0', handler)
+```
+
+### unhandle
+
+Unregisters all handlers with the given protocols
+
+`libp2p.unhandle(protocols)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| protocols | `Array<String>|String` | protocols to unregister |
+
+#### Example
+
+```js
+// ...
+libp2p.unhandle('/echo/1.0.0')
```

Comment:
We'll convert to an array so I think we should be more specific in examples of our preference, even though we support just passing a string. I'd like to gradually reduce the footprint of accepted params as it increases the likelihood of bugs.
```suggestion
libp2p.unhandle(['/echo/1.0.0'])
```